src/server.rs

```diff
@@ -108,15 +137,22 @@ impl Handler {
             socket,
             direction,
             serve_dir,
+            callback,
         })
     }
 
     /// Completes the handshake with the client and services the request.
-    pub fn handle(self) -> Result<()> {
-        match self.direction {
+    pub fn handle(mut self) -> Result<()> {
+        let callback = match self.callback.take() {
+            Some(l) => l,
+            None => Box::new(|| {}),
```

Comment:
Is there a case in normal operation where this would be `None`? Our `Handler` is moved into `Handler::handle`, therefore we won't ever call it again. 
```suggestion
            None => unreachable!(),
```